test/functional/feature_rbf.py

```diff
@@ -638,5 +656,129 @@ def test_replacement_relay_fee(self):
+
+        # clean up all evicted utxos / update wallet utxo state
+        self.wallet.get_utxo(txid=tx['txid'])
+        self.wallet.get_utxo(txid=replacement_tx['txid'])
+
+    def test_dynamic_inherited_signaling(self):
+        confirmed_utxos = [self.wallet.get_utxo(), self.wallet.get_utxo()]
+
+        # One opt-in parent transaction, one opt-out
+        parent_txs = list(map(lambda utxo: self.wallet.send_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=utxo,
+            sequence=BIP125_SEQUENCE_NUMBER if confirmed_utxos.index(utxo) == 0 else 0xffffffff,
+            fee_rate=Decimal('0.0001'),
+        ), confirmed_utxos))
+        optin_parent_tx, optout_parent_tx = parent_txs
+
+        # Craft a transaction that spends both parents so we can create one chain of descendant transactions
+        parent_utxos = list(map(lambda tx: self.wallet.get_utxo(txid=tx['txid']), parent_txs))
+        joined_tx_hex = self.create_multiple_input_self_transfer(parent_utxos, Decimal('0.0001'), sequence=0xffffffff)
+        joined_tx_txid = self.nodes[0].sendrawtransaction(joined_tx_hex)
+
+        # Only a single input needs to signal replaceability (whether explicitly or through inheritance) for the resulting transaction to signal
+        assert_equal(True, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Get the `joined_tx` utxo into our wallet so we can spend a chain of descendants from it
+        joined_tx = self.nodes[0].decoderawtransaction(joined_tx_hex)
+        self.wallet.scan_tx(joined_tx)
+        joined_utxo = self.wallet.get_utxo(txid=joined_tx_txid)
+
+        # Create a chain the size of `MAX_REPLACEMENT_LIMIT` spending `joined_tx` - the BIP125 Rule #5 imposed limit
+        tx = None
+        for _ in range(MAX_REPLACEMENT_LIMIT - 1):
+            tx = self.wallet.send_self_transfer(
+                from_node=self.nodes[0],
+                utxo_to_spend=joined_utxo if tx is None else self.wallet.get_utxo(txid=tx['txid']),  # a straight line of descendants
+                sequence=0xffffffff,
+                fee_rate=Decimal('0.0001'),
+            )
+            assert_equal(True, self.nodes[0].getmempoolentry(tx['txid'])['bip125-replaceable'])  # inherited
+
+        # Attempting to replace the opt-in parent transaction will now result in more than `MAX_REPLACEMENT_LIMIT`
+        # conflicting txns being evicted from the mempool. However, it (and all of its descendants) are still signaling replaceability.
+        # We would've expected that once `MAX_REPLACEMENT_LIMIT` is exceeded, the opt-in parent txn stops signaling
+        # replaceability, along with _all_ of its descendants.
+        entry = self.nodes[0].getmempoolentry(parent_txs[0]["txid"])
+        assert_greater_than(entry['descendantcount'], MAX_REPLACEMENT_LIMIT)
+        assert_equal(True, self.nodes[0].getmempoolentry(parent_txs[0]["txid"])['bip125-replaceable'])
```

Comment:
Minor suggestion, otherwise it is possibly unclear if 0 is opt-in and 1 is opt-out, or vice versa.
```suggestion
        assert_equal(True, self.nodes[0].getmempoolentry(optin_parent_tx["txid"])['bip125-replaceable'])
```